core/src/elo-connector/elo-connector.service.ts

```diff
@@ -0,0 +1,291 @@
+
+        if (isNcp && !winningTeam) return "Winning team must be specified if NCPing replays";
+
+        // Make sure we are considering replayIds in chronological order
+        replayIds.sort((r1, r2) => r1 - r2);
+
+        // Gather replays
+        const replayPromises = replayIds.map(async rId => this.roundRepository.findOneOrFail(rId));
+        const replays = await Promise.all(replayPromises);
+
+        // Check to make sure the winning team played in each replay
+        if (winningTeam) {
+            for (const replay of replays) {
+                if (replay.isDummy) continue; // Don't need to check dummy replays
+                // const teamsInReplay = (await
+                // replay.teamStats.loadItems()).map(tcs => tcs.team);
+                const teamsInReplay = replay.teamStats.map(tsl => tsl.teamName);
+                if (!teamsInReplay.includes(winningTeam.franchise.profile.title)) {
+                    this.logger.error(`The team \`${winningTeam.franchise.profile.title}\` did not play in replay with id \`${replay.id}\` (${teamsInReplay.join(" v. ")}), and therefore cannot be marked as the winner of this NCP. Cancelling process with no action taken.`);
+                    this.logger.warn(`Could not find team=${winningTeam.franchise.profile.title} on replay with id=${replay.id}, cannot mark as NCP`);
+                    throw new Error(`Could not find team=${winningTeam.franchise.profile.title} on replay with id=${replay.id}, cannot mark as NCP`);
+                }
+            }
+        }
+
+        // Set replays to NCP true/false and update winning team/color
+        for (const replay of replays) {
+            let newHomeWon: boolean = false;
+
+            // Handle NCPing/Un-NCPing with non-dummy/dummy replays
+            if (isNcp) {
+                if (!replay.isDummy) {
+                    newHomeWon = !replay.homeWon;
+                } else {
+                    newHomeWon = false;
+                }
+            } else if (!replay.isDummy) {
+                newHomeWon = !replay.homeWon;
+            }
+
+            // If Un-NCPing a dummy replay, just delete it
+            if (!isNcp && replay.isDummy) {
+                await this.roundRepository.delete(replay.id);
+            } else {
+                replay.homeWon = newHomeWon;
+                await this.roundRepository.save(replay);
+            }
+
+        }
+
+        // Magic happens here to talk to the ELO service
+        await this.sendReplaysToElo(replayIds, isNcp);
+
+        const outStr = `\`${replayIds.length === 1
+            ? `replayId=${replayIds[0]}`
+            : `replayIds=[${replayIds.join(", ")}]`
+        }\` successfully marked \`ncp=${isNcp}\`, ${winningTeam
+            ? `\`winningTeam=${winningTeam.franchise.profile.title}\``
+            : ""
+        } with updated elo, and all connected replays had their elo updated.`;
```

Comment:
Somewhat pedantic as well; why return a string here instead of building the message elsewhere?

Reply:
This one in particular was pulled straight from MLEDB. I only adjusted the field names to the variables I was using. I'm open to changing this to whatever we want. 
 
 The big change from MLEDB is that I pulled out the 'progressMessage' object, as I felt it was unnecessarily complicating things.

Reply:
This is now the only place in the method where a string is returned. We can decide on whether to change that later. https://app.clickup.com/t/2pdz0e8